service/models.py

```diff
@@ -168,6 +168,36 @@ def find_by_title(cls, title):
         logger.info("Processing title query for %s ...", title)
         return cls.query.filter(cls.title == title)
 
+    @classmethod
```

Comment:
These can be replaced with Promotion.find_by_fields({"promo_type": promotion.promo_type})